components/TextFields/src/ContainedInputView/private/MDCTextControlAssistiveLabelViewLayout.m

```diff
@@ -0,0 +1,175 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCTextControlAssistiveLabelViewLayout.h"
+
+@interface MDCTextControlAssistiveLabelViewLayout ()
+
+@property(nonatomic, assign) CGRect leftAssistiveLabelFrame;
+@property(nonatomic, assign) CGRect rightAssistiveLabelFrame;
+@property(nonatomic, assign) CGFloat calculatedHeight;
+
+@end
+
+@implementation MDCTextControlAssistiveLabelViewLayout
+
+#pragma mark Object Lifecycle
+
+- (instancetype)initWithWidth:(CGFloat)superviewWidth
+                  leftAssistiveLabel:(UILabel *)leftAssistiveLabel
+                 rightAssistiveLabel:(UILabel *)rightAssistiveLabel
+          assistiveLabelDrawPriority:
+              (MDCTextControlAssistiveLabelDrawPriority)assistiveLabelDrawPriority
```

Comment:
trying to follow along here: How does a client choose a different draw priority? We should have a test turning that knob. (we might already have this, I didn't see anything in the PR)

Reply:
No API associated with it ever got approved so my plan was going to be to just land it without exposing anything and then expose it if/when a client wanted something other than a priority of `.trailing`.

Reply:
There are actually a few things that I built solutions for that do not currently have any public APIs associated with them. This is one of them. These are the other two:
1. `density`. With my solution for density there is just a CGFloat property on the view that you can specify a value for between 0 and 1 inclusive where 0 is not dense (this is the default) and 1 is dense, meaning there is very little vertical padding between the various subviews.
2. `preferredContainerHeight`. One concern that I thought clients might have about these textfields was that their heights were affected by things like font, and there wasn't really a way to specify exactly how tall they should be, which iOS developers are pretty used to being able to do. To address that potential concern I came up with this notion of `preferredContainerHeight`. It's also a CGFloat. When it's set to a value the layout object tries to come up with a layout that results in the "container" of the textfield being exactly that many points tall. The container is the region in the outline for outlined textfields and the area of the filled background in filled textfields.

What should we do about these things that we don't immediately plan to make public? Should we throw them away? Or maybe we can bring them in and watch clients' behavior for a while and then later make the decision to delete them (this wouldn't be a breaking change in any way) or expose them? I think we should do the latter. I don't think there is any harm in bringing them in, even if it's temporary. It's also actually just easier for me and will probably result in things landing more quickly.

Reply:
Ok, lets leave as is. It may be worth adding a comment to explain this private feature not yet asked for by clinets. (you may have already done so and I missed this)